plugins/aver-hid/fu-aver-hid-device.c

```diff
@@ -102,10 +102,19 @@ fu_aver_hid_device_setup(FuDevice *device, GError **error)
 	if (!FU_DEVICE_CLASS(fu_aver_hid_device_parent_class)->setup(device, error))
 		return FALSE;
 
-	/* get the version and other properties from the hardware while open */
-	if (!fu_aver_hid_device_ensure_version(self, error))
+	/* using isp status requests as polling device requests */
+	if (!fu_aver_hid_device_poll(device, error))
 		return FALSE;
 
+	/* get the version from the hardware while open */
+	/* set a default version if the hardware is not support the version command, ignore the
+	 * error */
```

Comment:
if the hardware doesn't support the version command -- what happens? e.g. does the GetReport return with a failure, or is the report_id_custom_command different to expected etc. I think we need to handle this in a more generic way.

Reply:
We will add the version command to the new firmware of CAM520Pro3, but the old version does not support it. We hope it can function properly even if the version command is not supported.

Reply:
I created a branch named wip/aver/cam520pro3 on fwupd and pushed the commit.

Reply:
When the command is not supported, does fu_aver_hid_device_transfer() fail or fu_struct_aver_hid_res_device_version_validate()?

Reply:
Or is the version buf just NULs?